### DISCUSSION TITLE
Support for optional fields in Table Schema?

### DISCUSSION THREAD

**Discussion by roll** [frictionlessdata/frictionlessdata.io#653] [Ideas] [answered]
# Overview

Sometimes an user could prefer to have a schema with an optional field. It means that different data tables described by the same schema could either have or doesn't have some column completely.

---

_From @hredestig_

I thought of situations where the user e.g. may want to include a certain column, say, 'temperature', and if so, there would be constraints saying that it has to be numeric, but that specifying temperature is not strictly required and while the user now could get around this by simply including a column with all missing values, it would appear more natural to me to be allowed to not include that column at all.

---

_From @roll_

It could put some stress on implementations esp. while we still don't require field names to be unique so I recommend to be careful with this feature - think twice, check with different language implementators etc.

**Comment by Stephen-Gates** [comment]
An example would be the [GTFS specification](https://developers.google.com/transit/gtfs/reference/) which has a lot of optionality.

**Comment by roll** [comment, marked as answer]
Frictionless Framework support the `detector.schema_sync` argument that can be passed to `Resource/validate/etc` classed and functions - https://framework.frictionlessdata.io/docs/guides/framework/detector-guide#schema-sync

**Reply by thbar** [reply]
I needed to ensure a file with missing optional columns would be considered valid and ended up here.

The up-to-date link seems to be: https://v4.framework.frictionlessdata.io/docs/guides/framework/detector-guide#schema-sync

And TIL there is a `--schema-sync` option for the CLI too.

I now just wonder how to only "skip" optional columns, and not others.

**Reply by roll** [reply]
Hi @thbar,

currently, it's being discussed on the spec level. Can you please check if it resolved you case - https://github.com/frictionlessdata/specs/issues/367#issuecomment-1999256894
